hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestHoodieClientOnCopyOnWriteStorage.java

```diff
@@ -1114,6 +1114,16 @@ public void testClusteringWithSortColumns() throws Exception {
         .withClusteringTargetPartitions(0).withInlineClusteringNumCommits(1).build();
     testClustering(clusteringConfig);
   }
+  
+  @Test
+  public void testClusteringWithOneFilePerGroup() throws Exception {
+    HoodieClusteringConfig clusteringConfig = HoodieClusteringConfig.newBuilder().withClusteringMaxNumGroups(10)
+        .withClusteringMaxBytesInGroup(10) // set small number so each file is considered as separate clustering group
+        .withClusteringExecutionStrategyClass("org.apache.hudi.ClusteringIdentityTestExecutionStrategy")
```

Comment:
Can add a config

Reply:
This is just a unit test. Will provide another schedule clustering strategy as part of another PR to limit number of files per group.